src/gitflow/ui/GitflowWidget.java

```diff
@@ -178,4 +193,43 @@ private static String getDisplayableBranchTooltip(GitRepository repo) {
         }
         return text;
     }
+
+    @NotNull
+    private static String getWidgetID() {
+        return GitflowWidget.class.getName();
+    }
+
+    /**
+     * This method looks up the widget instance for a specific project
+     *
+     * @param project The project for which the widget instance should be looked up
+     * @return The widget instance for the provided project or null if no instance is available
+     */
+    @Nullable
+    public static GitflowWidget findWidgetInstance(@Nullable Project project) {
+        if (project != null) {
+            StatusBar statusBar = WindowManager.getInstance().getStatusBar(project);
+
+            if (statusBar != null) {
+                StatusBarWidget possibleWidget = statusBar.getWidget(getWidgetID());
+                if (possibleWidget instanceof GitflowWidget)
+                    return (GitflowWidget) possibleWidget;
+            }
+        }
+
+        return null;
+    }
+
+    /**
+     * Shows the action popup of this widget in the center of the provided frame. If there are no
+     * actions available for this widget, the popup will not be shown.
+     *
+     * @param frame The frame that will be used for display
+     */
+    public void showPopupInCenterOf(@NotNull JFrame frame) {
```

Comment:
I am not sure, if the content of this function should be wrapped into some kind of ```ApplicationManager.getApplication().invokeLater(/* ... */);```? I did not fully get why you are running the update method asynchronously. I tried it both ways and could not see a difference. So I sticked with the easy solution ;-)